src/ray/core_worker/transport/direct_task_transport.cc

```diff
@@ -281,8 +281,17 @@ void CoreWorkerDirectTaskSubmitter::RequestNewWorkerIfNeeded(
     return;
   }
 
-  auto lease_client = GetOrConnectLeaseClient(raylet_address);
   TaskSpecification &resource_spec = task_queue.front();
+  if (raylet_address == nullptr) {
+    // If no raylet address is given, try to find the best worker for our next lease
+    // request.
+    if (auto addr = lessor_picker_->GetBestNodeAddrForTask(resource_spec)) {
+      raylet_address = &addr.value();
```

Comment:
Is there a cleaner way to convert an `absl::optional<T>` to a `T*` after you have confirmed that the former has a value? I'm also happy to adopt a better "optional" pattern for `raylet_address` and/or the `LessorPicker` API in this PR, although I was planning on punting that until the spillback PR.